faucet/src/http.rs

```diff
@@ -211,35 +227,46 @@ async fn atomic_faucet_funding(
 
     ensure_funding_allowed(
         parachain_rpc,
-        account_id.clone(),
-        allowance_config,
+        &account_id,
+        &allowance_config,
         last_request_json,
         account_type.clone(),
     )
     .await?;
 
-    let mut transfers = vec![];
-    for AllowanceAmount { symbol, amount } in amounts.iter() {
-        let currency_id = CurrencyId::try_from_symbol(symbol.clone())?;
-        log::info!(
-            "AccountId: {}, Currency: {:?} Type: {:?}, Amount: {}",
-            account_id,
-            currency_id.symbol().unwrap_or_default(),
-            account_type,
-            amount
-        );
-        transfers.push(parachain_rpc.transfer_to(&account_id, *amount, currency_id));
+    if let Some(auth_url) = allowance_config.auth_url {
+        ensure_signature_exists(&auth_url, &account_id).await?;
     }
 
-    let result = futures::future::join_all(transfers).await;
+    // replace the previous (expired) claim datetime with the datetime of the current claim
+    update_kv_store(&kv, account_str.clone(), Utc::now().to_rfc2822(), account_type.clone())?;
```

Comment:
Can we have this be atomic with `ensure_funding_allowed`? Otherwise people might still be able to send multiple requests in quick succession (for example, if the auth check takes some time)